lein-finagle-clojure/src/leiningen/finagle_clojure.clj

```diff
@@ -33,15 +55,29 @@
     lein finagle-clojure scrooge :lint --help # shows available options for the linter
     lein finagle-clojure scrooge :lint -w # show linter warnings as well (warnings won't prevent compilation)"
   [project & options]
+  (doseq [f (classpath/get-classpath project)]
+    (pom/add-classpath f))
   (let [subtask (first options)
         project-root (:root project)
         source-path (get-in project [:finagle-clojure :thrift-source-path])
         raw-dest-path (get-in project [:finagle-clojure :thrift-output-path])]
     (if-not (and source-path raw-dest-path)
       (leiningen.core.main/info "No config found for lein-finagle-clojure, not compiling Thrift for" (:name project))
       (let [absolute-dest-path (->> raw-dest-path (io/file project-root) (.getAbsolutePath))
-            thrift-files (thrift-files project-root source-path)
-            scrooge-args (concat ["--finagle" "--skip-unchanged" "--language" "java" "--dest" absolute-dest-path] thrift-files)]
+            thrift-files (find-thrift-files project-root source-path)
+            include-filenames (->> thrift-files
+                                   (mapcat #(scrape-includes (slurp %)))
+                                   ;; Verify dependency file isn't already in the thrift source path
+                                   (filter #(every? (fn [^String thrift-file]
+                                                      (not (.endsWith thrift-file %)))
```

Comment:
For what it's worth, you can now use `clojure.string/ends-with?` instead of Java interop.